src/Functions/IFunction.cpp

```diff
@@ -250,7 +250,10 @@ ColumnPtr IExecutableFunction::executeWithoutSparseColumns(const ColumnsWithType
                                         : columns_without_low_cardinality.front().column->size();
 
             auto res = executeWithoutLowCardinalityColumns(columns_without_low_cardinality, dictionary_type, new_input_rows_count, dry_run);
-            auto keys = res->convertToFullColumnIfConst();
+            bool res_is_constant = isColumnConst(*res);
```

Comment:
When default implementation is used,  `Const(LowCardinality(...))` loses its constant property and becomes `LowCardinality(...)`, which in turn disables pruning with virtual columns.

Here is the fix.